types/types.go

```diff
@@ -34,15 +34,15 @@ type ComMatrix struct {
 }
 
 type ComDetails struct {
-	Direction string `json:"direction" yaml:"direction"`
-	Protocol  string `json:"protocol" yaml:"protocol"`
-	Port      int    `json:"port" yaml:"port"`
-	Namespace string `json:"namespace" yaml:"namespace"`
-	Service   string `json:"service" yaml:"service"`
-	Pod       string `json:"pod" yaml:"pod"`
-	Container string `json:"container" yaml:"container"`
-	NodeRole  string `json:"nodeRole" yaml:"nodeRole"`
-	Optional  bool   `json:"optional" yaml:"optional"`
+	Direction string `json:"direction" yaml:"direction" csv:"Direction"`
+	Protocol  string `json:"protocol" yaml:"protocol" csv:"Protocol"`
+	Port      int    `json:"port" yaml:"port" csv:"Port"`
+	Namespace string `json:"namespace" yaml:"namespace" csv:"Namespace"`
+	Service   string `json:"service" yaml:"service" csv:"Service"`
+	Pod       string `json:"pod" yaml:"pod" csv:"Pod"`
+	Container string `json:"container" yaml:"container" csv:"Container"`
+	NodeRole  string `json:"nodeRole" yaml:"nodeRole" csv:"Node Role"`
+	Optional  bool   `json:"optional" yaml:"optional" csv:"Optional"`
 }
 
 func ToCSV(m ComMatrix) ([]byte, error) {
```

Comment:
With this solution, should this func be updated and can we remove consts.CSVHeaders?

Reply:
I'm not sure, since this func converts a matrix to CSV and the solution allows converting a CSV to a matrix.